src/src/gd32f1x0/gpio.c

```diff
@@ -54,7 +54,7 @@ gpio_in_t gpio_in_setup(uint32_t pin, int32_t pull_up)
 
 uint8_t gpio_in_read(gpio_in_t g)
 {
-    return !!(GPIO_ISTAT(g.regs) & (g.bit));
+    return (GPIO_ISTAT(g.regs) & (g.bit));
```

Comment:
Why this is changed? That should always return only 0 or 1 so `!!` should be kept.

Reply:
Ah, I wasn't sure what !! did but your comment makes perfect sense.  I'll revert it

Reply:
You can change that to `return () != 0;` if makes more sense :)